pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -657,19 +688,12 @@ contract StablePhantomPool is
                     preJoinExitSupply,
                     preJoinExitInvariant,
                     currentAmp,
-                    balancesWithoutBpt,
-                    scalingFactors,
+                    balances,
+                    _dropBptItem(scalingFactors), // drop BPT to match balances
```

Comment:
I think this is unnecessary - we can do it at the place where we use the scaling factors, instead of here. And that way scaling factors will always be for the full registered set.

Reply:
Yeah, I tried to do it higher, where it would have made more sense.